pkg/provider/pods.go

```diff
@@ -179,3 +179,12 @@ func (p *Pod) ContainsIstioProxy() bool {
 	}
 	return false
 }
+
+func (p *Pod) CreatedByDeploymentConfig() bool {
+	for _, data := range p.ObjectMeta.GetOwnerReferences() {
+		if data.Kind != "ReplicationController" {
```

Comment:
A couple of issues here:
1. You return "false" as soon as the first "ownerReference" is not a ReplicationController. What if you have more than one ownerReference in the slice but the ReplicationController reference is the last one?
2. DeploymentConfig resources will create ReplicationControllers to deploy the pods. But you can also have ReplicationControllers in your cluster that have been deployed manually, without any DeploymentConfig. You need to return "true" only in case the owner of the pod is a ReplicationController whose owner is a DeploymentConfig. For example, if your pod's name is "X":
```
Pod X:
   ownerRerefences:
      - kind: ReplicationController
	  - name: rc1
	  
ReplicationController (mame: rc1)
   ownerRerefences:
      - kind: DeploymentConfig
```

Reply:
I agree and fixed it.